### PR TITLE
added documentation for entities

### LINKED ISSUE #1489: [Documentation] Custom entities
@shahednasser has you asked me to do, here is the issue about the entities.

We need to show how it is possible to add custom entities as well as a small word about the `BaseEntity`/`SoftDeletableEntity`

Let me know if you need anything :)

### PR DESCRIPTION
**Pull request by shahednasser**
Added documentation for entities and how to create them based on the PR #1315 

Closes #1489

### CODE REVIEW

File: docs/content/advanced/backend/entities.md [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+
+## Overview
+
+Entities in medusa represent tables in the database as classes. An example of this would be the `Order` entity which represents the `order` table in the database. Entities provide a uniform way of defining and interacting with data retrieved from the database.
+
+Aside from Medusa’s core entities, you can also create your own entities to use in your Medusa server. Custom entities must reside in the `src/models` directory of your Medusa server.
+
+Entities are TypeScript files and they are based on [Typeorm’s Entities](https://typeorm.io/entities) and use Typeorm decorators.
+
+All entities must extend either the `BaseEntity` or `SoftDeletableEntity` classes. The `BaseEntity` class holds common columns including the `id`, `created_at`, and `updated_at` columns. It also includes the method that takes care of automatically generating the ID field of your entity.
```
Comment (adrien2p) [inline docs/content/advanced/backend/entities.md:13]: The id generation utils has been moved to its own utility function 😊

Comment (adrien2p) [reply docs/content/advanced/backend/entities.md:13]: It is also not mandatory to extends those base classes but optional

File: docs/content/advanced/backend/entities.md [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+This will remove any errors that show up in your IDE related to experimental decorators.
+
+### Create the Entity
+
+To create an entity, create a TypeScript file in `src/models`. For example, here’s a `Post` entity defined in the file `src/models/post.ts`:
+
+```tsx
+import { BeforeInsert, Column, Entity, PrimaryColumn } from "typeorm";
+import { BaseEntity} from "@medusajs/medusa/dist/interfaces/models/base-entity";
+import { generateEntityId } from "@medusajs/medusa/dist/utils/generate-entity-id"
```
Comment (adrien2p) [inline docs/content/advanced/backend/entities.md:40]: Is it not exported from @medusajs/medusa ?

Comment (adrien2p) [reply docs/content/advanced/backend/entities.md:40]: Same for above?

File: docs/content/advanced/backend/entities.md [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+
+To generate an ID for your entity that matches the IDs generated for Medusa’s core entities, you should add a `BeforeInsert` event handler. Then, inside that handler use Medusa’s utility function `generateEntityId` to generate the ID. It accepts the ID as a first parameter and the prefix as a second parameter. The `Post` entity IDs will be of the format `post_<RANDOM>`.
+
+If you want the entity to also be soft deletable then it should extend `SoftDeletableEntity` instead:
+
+```tsx
+import { SoftDeletableEntity } from "@medusajs/medusa/dist/interfaces/models/soft-deletable-entity";
+
+@Entity()
+export class Post extends SoftDeletableEntity {
```
Comment (adrien2p) [inline docs/content/advanced/backend/entities.md:66]: Missing the closing }

File: docs/content/advanced/backend/entities.md [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+```
+
+:::
+
+You can access your custom entity data in the database in services or subscribers using the repository. For example, here’s a service that lists all posts:
+
+```tsx
+import { BaseService } from "medusa-interfaces";
+
+class PostService extends BaseService {
```
Comment (adrien2p) [inline docs/content/advanced/backend/entities.md:115]: Since the release will be soon, should we extend the TransactionBaseService?

**Review by adrien2p — COMMENTED** [review]
Small fixes 😊but its great

File: docs/content/advanced/backend/entities.md (hunk 3 above)
Comment (shahednasser) [reply docs/content/advanced/backend/entities.md:66]: I know this is just to showcase where the change should be made but I guess I should make it clearer

File: docs/content/advanced/backend/entities.md (hunk 4 above)
Comment (shahednasser) [reply docs/content/advanced/backend/entities.md:115]: Is this in the PR #1315 or in another PR? I didn't see any reference of it

Comment (adrien2p) [reply docs/content/advanced/backend/entities.md:117]: It comes from these one https://github.com/medusajs/medusa/pull/1344

**Comment by shahednasser** [comment]
@adrien2p thanks for reviewing it! I made all changes as you pointed out but regarding the import of the classes, I think I'll wait until you look into it, correct? Let me know if I should make any changes

**Comment by shahednasser** [comment]
@adrien2p is this all good? I saw that the PR was merged but there are no changes to the import/export

**Comment by adrien2p** [comment]
@shahednasser gosh, forget that... i will add it tomorrow

**Comment by shahednasser** [comment]
@adrien2p hope you feel better!

File: docs/content/advanced/backend/entities.md [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+:::
+
+You can access your custom entity data in the database in services or subscribers using the repository. For example, here’s a service that lists all posts:
+
+```tsx
+import { TransactionBaseService } from "medusa-interfaces";
+
+class PostService extends TransactionBaseService {
+  constructor({ postRepository, manager }) {
+    super();
```
Comment (adrien2p) [inline docs/content/advanced/backend/entities.md:119]: Issue: Missing super argument

**Review by adrien2p — CHANGES_REQUESTED** [review]
LGTM! Just one issue

File: docs/content/advanced/backend/entities.md (hunk 5 above)
Comment (adrien2p) [reply docs/content/advanced/backend/entities.md]: 👌

**Review by adrien2p — APPROVED** [review]
LGTM! 🚀

**Pull request closed without merging**